code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -286,3 +286,164 @@
+	UnregisterSignal(hunter_mark_target, COMSIG_PARENT_PREQDELETED)
+	var/datum/action/xeno_action/psychic_trace/psychic_trace_set_target = owner.actions_by_path[/datum/action/xeno_action/psychic_trace]
+	psychic_trace_set_target.psychic_trace_target = null //Nullify psychic trace's target and clear the var
+	hunter_mark_target = null //Nullify hunter's mark target and clear the var
+
+// ***************************************
+// *********** Psychic Trace
+// ***************************************
+/datum/action/xeno_action/psychic_trace
+	name = "Psychic Trace"
+	action_icon_state = "toggle_queen_zoom"
+	mechanics_text = "Psychically ping the creature you marked, letting you know its direction, distance and location, and general condition."
+	plasma_cost = 1 //Token amount
+	keybind_signal = COMSIG_XENOABILITY_PSYCHIC_TRACE
+	cooldown_timer = HUNTER_PSYCHIC_TRACE_COOLDOWN
+	var/mob/living/psychic_trace_target
+
+/datum/action/xeno_action/psychic_trace/can_use_action(silent = FALSE, override_flags)
+	. = ..()
+
+	if(!psychic_trace_target)
+		if(!silent)
+			to_chat(owner, "<span class='xenowarning'>We have no target we can trace!</span>")
+		return FALSE
+
+	if(psychic_trace_target.z != owner.z)
+		if(!silent)
+			to_chat(owner, "<span class='xenowarning'>Our target is too far away, and is beyond our senses!</span>")
+		return FALSE
+
+
+/datum/action/xeno_action/psychic_trace/action_activate()
+
+	var/mob/living/carbon/xenomorph/hunter/X = owner
+
+	to_chat(X, "<span class='xenodanger'>We sense our quarry <b>[psychic_trace_target]</b> is currently located in <b>[AREACOORD_NO_Z(psychic_trace_target)]</b> and is <b>[get_dist(X, psychic_trace_target)]</b> tiles away. It is <b>[calculate_mark_health(psychic_trace_target)]</b> and <b>[psychic_trace_target.status_flags & XENO_HOST ? "impregnated" : "barren"]</b>.</span>")
+	X.playsound_local(X, 'sound/effects/ghost2.ogg', 10, 0, 1)
+
+	var/obj/screen/arrow/arrow_hud = new /obj/screen/arrow/hunter_mark_arrow //Prepare the tracker object and set its parameters
+	//Prepare the tracker object and set its parameters
+	arrow_hud.add_hud(X, psychic_trace_target) //set the tracker parameters
+	arrow_hud.color = COLOR_ORANGE
+	arrow_hud.process() //Update immediately
+
+	add_cooldown()
+
+	return succeed_activate()
+
+///Where we calculate the approximate health of our trace target
+/datum/action/xeno_action/psychic_trace/proc/calculate_mark_health(mob/living/target)
+
+	if(target.stat == DEAD)
+		return "deceased"
+
+	var/percentage = round(target.health * 100 / target.maxHealth)
+	switch(percentage)
+		if(100 to INFINITY)
+			return "in perfect health"
+		if(76 to 99)
+			return "slightly injured"
+		if(51 to 75)
+			return "moderately injured"
+		if(26 to 50)
+			return "badly injured"
+		if(1 to 25)
+			return "severely injured"
+		if(-51 to 0)
+			return "critically injured"
+		if(-99 to -50)
+			return "on the verge of death"
+		else
+			return "deceased"
```

Comment:
should this happen? 
You might want to make this something else in case of bugs